cmd/kubelet/kubelet.go

```diff
@@ -99,6 +120,26 @@ func main() {
 		die(err)
 	}
 
+	// We must enforce flag precedence by re-parsing the command line into the new object.
+	// This is necessary to preserve backwards-compatibility across binary upgrades.
+	// See issue #56171 for more details.
+	// We use a throwaway kubeletFlags and a fake global flagset to avoid double-parses,
+	// as some Set implementations accumulate values from multiple flag invocations.
+	globalfs := newFlagSetWithGlobals()
+	newfs := newFakeFlagSet(globalfs)
+
+	// register throwaway KubeletFlags
+	tmp := options.NewKubeletFlags()
+	tmp.AddFlags(newfs)
+
+	// register new KubeletConfiguration
+	options.AddKubeletConfigFlags(newfs, kubeletConfig)
+
+	// re-parse flags
+	if err := parseFlagSet(newfs, cliArgs); err != nil {
```

Comment:
should probably skip `glog.V(2).Infof("FLAG: --%s=%q"` on reparse... looks silly to echo them a second time, especially when the second echo shows empty fake global flags

Reply:
Done